.circleci/config.yml

```diff
@@ -0,0 +1,27 @@
+
+version: 2
+jobs:
+  build:
+    working_directory: ~/graft
+    parallelism: 1
+    docker:
+    - image: circleci/node:8.11.4-stretch
+
+    steps:
+      - run: curl -sSL https://get.haskellstack.org/ | sh
+      - checkout
+
+      - restore_cache:
+          keys:
+          - cv-graft-cache-v1-dep-{{ .Branch }}-{{ checksum "stack.yaml" }}
```

Comment:
Minor: We should probably also cache based on checksum of `graft.cabal` as compiler options, etc. can invalidate cache, e.g. https://github.com/collegevine/rollbar/blob/08d7405292d71b94ea3c0562716330e77cfb93d8/.circleci/config.yml#L27

Reply:
At first I also thought to include the cabal file, but then I thought, is it really required?
The dependency versions are pinned by `stack.yaml`, they're absent from the cabal file, and therefore, changes to cabal file should not invalidate the cache. If I add a new dependency to the cabal file, then it will be added to `~/.stack` and cached for next time, but I cannot change the cabal file in a way that would require erasing the cache. Makes sense?

Reply:
Good point. I was thinking of changing GHC options, `default-extensions`, etc. in Cabal file should invalidate the cache as the build output would change, right?

Reply:
If the build output would change, the GHC would just rebuild. Look at it this way: when you change flags on your local, you don't erase the `.stack-word` directory, do you?

Reply:
I guess the benefit is that if you have a branch + `stack.yaml` based cache and change the Cabal file, you can’t differentiate between the two caches. Not a huge loss I guess and this is how our fallbacks work: specific, then branch, then `master`.